qiime2/sdk/result.py

```diff
@@ -409,14 +446,24 @@ def get_annotation(self, name):
 
         raise KeyError(f'No Annotation with name: "{name}" was found.')
 
-    # TODO: add support to filter by type
-    # once additional annotation types are added in 7.1
-    def iter_annotations(self):
+    def iter_annotations(self, filter_by_type=None):
         """Constructs an iterable containing all Annotations associated with
         the Result object.
         """
         self._validate_annotation_support()
-        yield from self._annotations.values()
+
+        if filter_by_type is None:
+            yield from self._annotations.values()
+            return
+
+        if filter_by_type not in ANNOTATION_TYPE_LIST:
+            raise ValueError(f'Unknown annotation type: "{filter_by_type}". '
+                             'Supported annotation types are: '
+                             f'{ANNOTATION_TYPE_LIST}')
+
+        for annotation in self._annotations.values():
+            if getattr(annotation, 'annotation_type') == filter_by_type:
+                yield annotation
```

Comment:
```suggestion

        if filter_by_type is None:
            yield from self._annotations.values()
        elif filter_by_type not in ANNOTATION_TYPE_LIST:
            raise ValueError(f'Unknown annotation type: "{filter_by_type}". '
                             'Supported annotation types are: '
                             f'{ANNOTATION_TYPE_LIST}')
        else:
            for annotation in self._annotations.values():
                if getattr(annotation, 'annotation_type') == filter_by_type:
                    yield annotation
```

I think it's easier to read like this because it's just an if/elif/else without the sneaky return and whatnot, but this is an opinion thing.

Reply:
no this is totally fair- i don't have strong opinions on it, this was just the way i ended up thinking through this section; this is definitely more readable.